Framework/Geometry/src/Instrument/SampleEnvironmentSpecParser.cpp

```diff
@@ -222,16 +223,18 @@ Mantid::Geometry::SampleEnvironmentSpecParser::parseComponent(
   }
   ShapeFactory factory;
   auto comp = factory.createShape(geometry);
-  comp->setID(element->getAttribute("id"));
   auto materialID = element->getAttribute("material");
   auto iter = m_materials.find(materialID);
+  Kernel::Material mat;
   if (iter != m_materials.end()) {
-    comp->setMaterial(iter->second);
+    mat = iter->second;
   } else {
     throw std::runtime_error("SampleEnvironmentSpecParser::parseComponent() - "
                              "Unable to find material with id=" +
                              materialID);
   }
+  comp->setID(element->getAttribute("id"));
+  comp->setMaterial(mat);
```

Comment:
Why the change from `comp->setMaterial(iter->second);`? It doesn't look like there is a reason for `mat` to be in function scope.